lib/core/network/protocols/httpwsProtocol.js

```diff
@@ -636,6 +623,160 @@ class HttpWsProtocol extends Protocol {
+      || httpStream.stream._readableState.ended
+    ) {
+      response.writeStatus('200 OK');
+      response.end();
+      return;
+    }
+
+    // Remove Content-Length header to avoid conflic with Transfer-Encoding header
+    request.response.setHeader('Content-Length', null);
+
+    // Send Headers in one go
+    response.cork(() => {
+      this.httpWriteRequestHeaders(request, response, message);
+
+      if (streamSizeFixed) {
+        response.writeHeader(HTTP_HEADER_CONTENT_LENGTH, Buffer.from(httpStream.totalBytes.toString()));
+      }
+      else {
+        response.writeHeader(HTTP_HEADER_TRANSFER_ENCODING, CHUNKED);
+      }
+    });
+
+    httpStream.stream.on('data', chunk => {
+      // Make a copy of the array buffer
+      const arrayBuffer = chunk.buffer.slice(chunk.byteOffset, chunk.byteOffset + chunk.byteLength);
+
+      const arrayBufferOffset = response.getWriteOffset();
+
+      let backpressure = false;
+
+      /**
+       * Switch method of writing data to the response
+       * based on the size of the stream.
+       * If the stream has a fixed size we can use the tryEnd method
+       */
+      if (streamSizeFixed) {
+        const [ success, done ] = response.tryEnd(arrayBuffer, httpStream.totalBytes);
+        backpressure = ! success;
+
+        if (done) {
+          httpStream.stream.destroy();
+          return;
+        }
+      }
+      else {
+        const success = response.write(arrayBuffer);
+
+        backpressure = ! success;
+      }
+
+      // In case of backpressure we need to wait for drainage before sending more data
+      if (backpressure) {
+        response.arrayBufferOffset = arrayBufferOffset;
+        response.arrayBuffer = arrayBuffer;
+        httpStream.stream.pause();
+
+        /**
+         * When the stream is drained we can resume sending data,
+         * only if the is no backpressure after we wrote the missing chunk data.
+         */
+        response.onWritable(offset => {
+          if (! streamSizeFixed && offset - response.arrayBufferOffset === 0) {
+            httpStream.stream.resume();
+            return true;
+          }
+
+          let retryBackpressure = false;
+          const remainingChunkData = response.arrayBuffer.slice(offset - response.arrayBufferOffset);
+
+          if (streamSizeFixed) {
+            const [ success ] = response.tryEnd(remainingChunkData, httpStream.totalBytes);
+
+            retryBackpressure = ! success;
+          }
+          else {
+            const success = response.write(remainingChunkData);
+
+            retryBackpressure = ! success;
+          }
+
+          if (! retryBackpressure) {
+            httpStream.stream.resume();
+          }
+
+          return ! retryBackpressure;
+        });
+      }
+    });
+
+    
+    httpStream.stream.on('end', () => {
+      response.end();
+      httpStream.stream.destroy();
+    });
+    
+
+    httpStream.stream.on('error', err => {
```

Comment:
What happen to the response if the user stream is destroyed? Shouldn't we `end` it with an error?

Reply:
Yes you are right I forgot to end the stream in case of stream failure